lib/manageiq/session.rb

```diff
@@ -0,0 +1,48 @@
+require "extensions/session_extension"
+
+module ManageIQ
+  module Session
+    FakeRequest = Struct.new(:env)
+
+    def self.store
+      @store ||= fetch_store_from_rails
+    end
+
+    def self.store=(session_store)
+      adapter_klass = "ManageIQ::Session::#{session_store.to_s.camelize}Adapter".safe_constantize
+      raise ArgumentError, "invalid session store adapter: #{session_store.inspect}" unless adapter_klass
+
+      configure_session_store(adapter_klass.new)
+
+      @store = nil
+    end
+
+    def self.configure_session_store(adapter)
+      Vmdb::Application.config.session_store(*adapter.session_store_config_args)
+      msg = "Using session_store: #{Vmdb::Application.config.session_store}"
+      _log.info("MIQ(SessionStore) #{msg}")
+      puts "** #{msg}" if !Rails.env.production? && adapter.type != :mem_cache_store
+    end
+
+    def self.revoke(session_ids)
+      store.delete_sessions(session_ids)
+    end
+
+    def self.fake_request
+      FakeRequest.new({})
```

Comment:
also super minor, but should we do this?
```suggestion
      @fake_request ||= FakeRequest.new({})
```

or even better, maybe at the top just define

```ruby
FakeRequest = Struct.new(:env).new
```

Then the callers just use `ManageIQ::Session::FakeRequest` directly.

Reply:
There was a place (that I can't think of at the moment) where `FakeRequest` was getting manipulated, so I didn't want keep that around in a class instance variable.